pkgs/sketch_pad/lib/editor/editor.dart

```diff
@@ -218,11 +242,29 @@ class _EditorWidgetState extends State<EditorWidget> implements EditorService {
 
     _updateCodemirrorMode(darkMode);
 
-    return HtmlElementView(
-      key: _elementViewKey,
-      viewType: _viewType,
-      onPlatformViewCreated: (id) =>
-          _platformViewCreated(id, darkMode: darkMode),
+    return FocusableActionDetector(
+      focusNode: _focusNode,
+      onFocusChange: (isFocused) {
+        // If focus is entering or leaving, convey this to CodeMirror.
+        if (isFocused) {
+          codeMirror?.focus();
+        } else {
+          codeMirror?.getInputField().blur();
+        }
+      },
+      shortcuts: {
+        // Add Esc and Shift+Esc as shortcuts for focus to leave editor.
```

Comment:
Do we need this? I wonder what the expectations are wrt focus traversal and text editors. I think that if the focus ever escapes the text editor, people would expect the normal focus traversal until the focus was again captured by the text editor. But do users expect to have a keybinding that lets them move the focus out of the text editor?

Reply:
I think we need some key binding like this as otherwise users have no easy way for focus traversal to escape CodeMirror with just the keyboard.

Reply:
I think I'll remove these for now, as that still leaves us in a better spot than currently, then investigate adding something like the `Esc+Tab` that CodeMirror 6 uses as it is likely what developers are becoming familiar with.